enhancements/baremetal/continuous-hw-introspection.md

```diff
@@ -0,0 +1,147 @@
+and become aware of any failures seen on the servers. This information can be used
+to report current status of hardware to the user and/or glean metrics about the
+health of the deployment.
+
+At the end of 4.4, a Baremetal IPI deployment does not have continuous hardware introspection
+running on its master or worker nodes. The underlying provisioning service already has the
+capability to perform hardware inspection on each of the servers periodically. The
+baremetal-operator (BMO) then reads this information and updates the "status" field of the
+BareMetalHost CRD. Please refer to [1] for the latest BaremetalHost CRD.
+
+## Motivation
+
+This enhancement will allow the user to get continuous updates on all the baremetal
+servers in their Metal3 deployment providing useful information about the hardware
+capabilities of the servers.
+
+### Goals
+
+This enhancement will allow the baremetal host CR associated with each baremetal
+server in a metal3 deployment to contain up-to-date information about its hardware and
+operational status of a server.
+
+### Non-Goals
+
+This proposal does not include any changes to the BareMetalHost CRD.
+
+## Proposal
+
+This enhancement proposes to add a new DaemonSet started by the machine-api-operator
+only for the BareMetal platform. This DaemonSet would run a new container image
+(ironic-hardware-inventory-recorder) on all master and worker nodes. With the new
+addition, it would become possible to have detailed information regarding the
+hardware connected to both master and worker nodes in a metal3 deployment and not
+just when the worker nodes are provisioned. This also allows the user to keep
+track of any changes in hardware capabilities of servers in a metal3 deployment.
+
+Additional details regarding enhancements made to the underlying service required
+for this feature can be found in [3].
+
+### User Stories [optional]
+
+#### Story 1
+
+As a user, I would like to know the current status of the hardware and its capabilities
+for each of the baremetal servers in my Baremetal IPI deployment.
+
+#### Story 2
+
+As a user, I would like to know of any changes in hardware status and capabalities of each
+baremetal server while it is part of my Baremetal IPI deployment.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Continuous hardware introspection involves periodically reading information about baremetal
+server hardware by talking to the server directly. This information includes the details
+details about devices connected to each server like its CPUs, NICs and storage devices.
+The actual task of performing hardware introspection is provided by a service that needs
+to run on all master and worker nodes. The baremetal-operator polls this service for
+hardware introspection data on all baremetal hosts when it is not busy reconciling. The
+baremetal-operator then updates the status on their respective BareMetalHost CRs.
```

Comment:
I think these last two sentences are conflating two different things.
AIUI the IPA agent container will run as a DaemonSet on all hosts. It pushes data back to ironic-inspector service (running within the baremetal-operator pod).
The baremetal-operator will (but currently does not) poll the ironic-inspector service to update the CRs.